tests/unit_tests/steps/test_archive_objects.py

```diff
@@ -82,20 +91,24 @@ def test_for_library(self, stub_tool_box,
                    match="_metric_send_conn not set, cannot send metrics"):
             archive_objects(config=config,
                             output_fpath=config.build_output / 'mylib.a')
-        assert call_list(fake_process) == [ar_command]
+
+        assert call_list(fake_process) == [help_command, ar_command]
 
         # ensure the correct artefacts were created
         assert config.artefact_store[ArtefactSet.OBJECT_ARCHIVES] == {
             None: {str(config.build_output / 'mylib.a')}}
 
-    def test_incorrect_tool(self, stub_tool_box):
+    def test_incorrect_tool(self, stub_tool_box, monkeypatch):
         """
         Test that an incorrect archive tool is detected.
         """
         config = BuildConfig('proj', stub_tool_box)
-        cc = stub_tool_box.get_tool(Category.C_COMPILER, config.mpi, config.openmp)
-        # And set its category to be AR
-        cc._category = Category.AR
+        cc = stub_tool_box.get_tool(Category.C_COMPILER, config.mpi,
+                                    config.openmp)
+        # And set its category to be AR. Use monkeypatch since the compiler
```

Comment:
Could it be worth adding the link to the documentation here to differentiate between general monkey patching and pytest monkey patching, as well as helping developers understand what it involves ? 

```suggestion
        # And set its category to be AR. Use monkeypatch (https://docs.pytest.org/en/6.2.x/monkeypatch.html) since the compiler
```